src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/instances/x64/softmax.cpp

```diff
@@ -0,0 +1,168 @@
+                          ov::PartialShape{-1, -1},
+                          {// target static shapes
+                           ov::Shape{10, 1},
+                           ov::Shape{15, 15},
+                           ov::Shape{10, 5},
+                           ov::Shape{15, 15}}},
+     0},
+    {ov::test::InputShape{// dynamic shape
+                          ov::PartialShape{ov::Dimension{1, 100}, ov::Dimension{1, 100}, -1},
+                          {// target static shapes
+                           ov::Shape{10, 10, 10},
+                           ov::Shape{10, 10, 1},
+                           ov::Shape{10, 5, 10},
+                           ov::Shape{10, 10, 1}}},
+     1},
+    {ov::test::InputShape{// dynamic shape
+                          ov::PartialShape{{1, 10}, 10},
+                          {// target static shapes
+                           ov::Shape{7, 10},
+                           ov::Shape{10, 10}}},
+     0},
+};
+
+const std::vector<SoftMaxConfig> unsupportedConfigsFP32{
+    // Static shapes
+    {ov::test::InputShape{ov::PartialShape{5, 5, 5, 5, 5, 5}, {ov::Shape{5, 5, 5, 5, 5, 5}}}, 0},
+    {ov::test::InputShape{ov::PartialShape{5, 5, 5, 5, 5, 5}, {ov::Shape{5, 5, 5, 5, 5, 5}}}, 1},
+    {ov::test::InputShape{ov::PartialShape{5, 5, 5, 5, 5, 5}, {ov::Shape{5, 5, 5, 5, 5, 5}}}, 2},
+    {ov::test::InputShape{ov::PartialShape{5, 5, 5, 5, 5, 5}, {ov::Shape{5, 5, 5, 5, 5, 5}}}, 3},
+    {ov::test::InputShape{ov::PartialShape{5, 5, 5, 5, 5, 5}, {ov::Shape{5, 5, 5, 5, 5, 5}}}, 4},
+    {ov::test::InputShape{ov::PartialShape{5, 5, 5, 5, 5, 5}, {ov::Shape{5, 5, 5, 5, 5, 5}}}, 5},
+    // Dynamic shapes
+    {ov::test::InputShape{// dynamic shape
+                          ov::PartialShape{-1, -1, -1, -1, -1, -1},
+                          {// target static shapes
+                           ov::Shape{5, 5, 5, 5, 5, 5},
+                           ov::Shape{7, 7, 7, 7, 7, 7},
+                           ov::Shape{5, 5, 5, 5, 5, 5}}},
+     4},
+    {ov::test::InputShape{// dynamic shape
+                          ov::PartialShape{{1, 10}, 5, 5, 5, 5, 5},
+                          {// target static shapes
+                           ov::Shape{5, 5, 5, 5, 5, 5},
+                           ov::Shape{7, 5, 5, 5, 5, 5}}},
+     4},
+};
+
+const auto NotOptimizedParams = testing::Combine(testing::Values(ElementType::f32),
+                                                 testing::ValuesIn(notOptimizedConfigsFP32),
+                                                 testing::Values(ov::test::utils::DEVICE_CPU),
+                                                 testing::Values(notOptimizedCPUSpec),
+                                                 testing::Values(cpu_f16_plugin_config));
+
+INSTANTIATE_TEST_SUITE_P(smoke_SoftMax_CPU_FP16,
+                         SoftMaxLayerCPUTest,
+                         NotOptimizedParams,
+                         SoftMaxLayerCPUTest::getTestCaseName);
```

Comment:
yes, this is the verbose 
`ov_cpu_verbose,exec,cpu,Softmax_20175:Softmax:Default,ref_any, src:0:f16::blocked:ab::f0:7x10 dst:0:f16::blocked:ab::f0:7x10,,0.02527ms`

Reply:
It looks like this not optimized params are misunderstood. Long time ago the optimized Softmax implementation in oneDNN supported only special cases of the layout and Softmax axis combination. The OV team extended this optimized Softmax implementation to support some other configurations. But now it looks like the oneDNN SoftMax primitive has no such limitations any more. Thus here should be only one parameters set including the existing "optimized" and "non optimized" params and it should be always expected that any of the optimized implementations is selected, e.i. jit_avx512, jit_avx2 or jit_sse, of course depending on the data type. So please rearrange the tests so that they always test the optimized implementations. It has no point to test the reference as it's never expected to be used.